SDCTA/src/components/auth/SignUpPage.tsx

```diff
@@ -0,0 +1,291 @@
+import { useState } from 'react';
+import { createUserWithEmailAndPassword,updateProfile } from "firebase/auth";
+import './auth.css';
+import { auth } from "../../firebase-config";
+import {registerUser} from "../../api/auth";
+import {signUpErrorHandler} from "../../error_handling/auth-errors"
+import {useNavigate} from 'react-router-dom';
+
+
+import { GoogleAuthProvider, getAuth } from "firebase/auth";
+import { signInWithRedirect, getRedirectResult} from "firebase/auth";
+
+
+export const SignUpPage = () => {
+
+  const [userDisplayName, setUserDisplayName] = useState("");
+  const [registerEmail, setRegisterEmail] = useState("");
+  const [registerPassword, setRegisterPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [agreedTerms,setAgreedTerms] = useState(false);
+  const navigate = useNavigate();
+
+  const [unknownError, setUnknownError] = useState("");
+  const [passwordError,setPasswordError] = useState("");
+  const [confirmError, setConfirmError] = useState("");
+  const [emailError,setEmailError] = useState("");
+  const [nameError,setNameError] = useState("");
+
+  const register = async () => {
+    try {
+
+      //make sure display name is set.
+      if (!(confirmPassword === registerPassword)) {
+        throw Error("no-match");
+      }
+
+      if (userDisplayName === "") {
+        throw Error("empty-name")
+      }
+
+      if (!(agreedTerms)) {
+        throw Error("no-terms");
+      }
+
+      const userCredential = await createUserWithEmailAndPassword(
+        auth,
+        registerEmail,
+        registerPassword
+      );
+
+      await updateProfile(userCredential.user, {
+        displayName: userDisplayName
+      });
+
+      await registerUser(userCredential);
+
+      navigate("/");
+
+    } catch (error) {
+
+      if (error instanceof Error) {
+        const errorMessage = await signUpErrorHandler(error);
+
```

Comment:
Same thing with InputErrorType I mentioned above can be applied here, you can define that type in the auth-errors file you have.